File: 백준/Bronze/2851. 슈퍼 마리오/슈퍼 마리오.py
arr = []
for _ in range(10):
    arr.append(int(input()))

ans = 0
#완전탐색 생각
for i in arr:
    ans += i
    if ans >= 100:
        if abs(ans - 100) > abs((ans - i) - 100): #무엇이 더 100이랑 가까운지 구하기
            ans -= i
        break

print(ans)

File: 백준/Bronze/2851. 슈퍼 마리오/슈퍼 마리오.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int[] arr = new int[10];
        for (int i = 0; i < 10; i++) {
            arr[i] = Integer.parseInt(br.readLine());
        }
        int sum = 0;
        int ans = 0;
        int diff = Integer.MAX_VALUE;
        for (int i = 0; i < 10; i++) {
            sum += arr[i];
            if(diff >= Math.abs(100 - sum)){
                ans = sum;
                diff = Math.abs(100 - sum);
            }
        }
        System.out.println(ans);
    }
}